### DISCUSSION TITLE
avoid inherited class detection in YOLOv5 inference

### DISCUSSION THREAD

**Discussion by baleris** [ultralytics/yolov5#6425] [Q&A]
@glenn-jocher
i have a trained model on custom dataset which is detecting 5 classes from an image. But i see there were some bounding boxes detected inside another detected bounding box.
For example, model detects a class-1 object which is having a size of around 100x100(bbox) and i see another class-2 of size 10x20(bbox) detected inside the area of class-1 detected bounding box. Also i see class-1 may re appear inside the already detected area of class-1 i.e i may have another smaller object of same class can appear inside the larger class already detected.

I would like to consider only outer larger bounding box which is detected by model and want to avoid any classes detected inside that area. Is there any possibility in general.py or any other configuration i can modify to achieve this ?

I tried these agnostic & multi_label options(below), wouldn't worked for me.
#Update inference default to multi_label=False #2252
#Limit bounding boxes #2248

**Comment by glenn-jocher** [comment]
@baleris you can try multi_label, agnostic and iou_thresh settings during inference.

**Reply by baleris** [reply]
i tried the multi_label = False , agnostic = True and my iou_thresh is 0.45 . i tried by changing different combinations like keeping both True and False couldn't see any difference.
Even though we have multi_label = False why its detecting multiple different class objects inside a larger class bounding box ?

**Reply by glenn-jocher** [reply]
@baleris multi_label affects detections on multiple classes on the same exact bounding box.

**Reply by baleris** [reply]
@glenn-jocher thanks for clarifying this .... is there any alternative way, either by change in code in general.py can help me to resolve this problem ?

**Reply by glenn-jocher** [reply]
@baleris like I said agnostic=True and vary iou_thresh. At some iou_thresh no overlap at all will be tolerated.

**Reply by baleris** [reply]
in my case the identified two classes are not having approx. same bounding boxes. Like in this screenshot, i want to avoid "Class-2" detected inside "Class-1".
![p1](https://user-images.githubusercontent.com/71070130/151195241-b5eda266-f934-45a2-9792-e8096c6c060c.JPG)
